### PR TITLE
WIP: HashAggregateExec

### PR DESCRIPTION
**Pull request by morpheyesh**
This is WIP PR. 
There isn't much added really. 
I am not entirely sure how to handle `AggregateExpr` in `to_proto`. 
Thanks for the help!

(Issue Link: #427 & #461 )

### CODE REVIEW

**Comment by andygrove** [comment]
I created a PR against Arrow to add `as_any` to `AggregateExpr` - https://github.com/apache/arrow/pull/9438

**Comment by morpheyesh** [comment]
Perfect. Thanks! I will finish this up.

File: rust/ballista/src/serde/physical_plan/to_proto.rs [hunk 1]
```diff
@@ -251,6 +256,30 @@ impl TryInto<protobuf::PhysicalPlanNode> for Arc<dyn ExecutionPlan> {
     }
 }
 
+impl TryInto<protobuf::LogicalExprNode> for Arc<dyn AggregateExpr> {
+    //need to fix!
+    type Error = BallistaError;
+
+    fn try_into(self) -> Result<protobuf::LogicalExprNode, Self::Error> {
```
Comment (andygrove) [inline rust/ballista/src/serde/physical_plan/to_proto.rs:263]:
```suggestion
    fn try_into(self) -> Result<protobuf::LogicalExprNode, Self::Error> {
        let aggr_function = if self.as_any().downcast_ref::<Avg>().is_some() {
            Ok(protobuf::AggregateFunction::Avg.into())
        } else if self.as_any().downcast_ref::<Sum>().is_some() {
            Ok(protobuf::AggregateFunction::Sum.into())
        } else {
            Err(BallistaError::NotImplemented(format!("Aggregate function not supported: {:?}", self)))
        }?;

        let expressions: Vec<protobuf::LogicalExprNode> = self.expressions()
            .iter()
            .map(|e| e.clone().try_into())
            .collect::<Result<Vec<_>, BallistaError>>()?;

        Ok(protobuf::LogicalExprNode {
            expr_type: Some(protobuf::logical_expr_node::ExprType::AggregateExpr(
                Box::new(protobuf::AggregateExprNode {
                    aggr_function,
                    expr: Some(Box::new(expressions[0].clone())),
                }),
            )),
        })
    }```

**Comment by andygrove** [comment]
@morpheyesh The Arrow PR got merged already so you can update the Arrow dependency hash to `8626325` to pick up the change.

**Comment by morpheyesh** [comment]
Completed in #497. Closing this :)

**Pull request closed without merging**